### PR TITLE
Fix recreate vault import accounts

### LINKED ISSUE #1105: The imported account is automatically deleted
If I create a new account and then import an old account with a private key, let me use the old account to send some tokens to the new account, then switch to the new account, I will be prompted to set the password. When I set the password, I found the switch account. The imported account was deleted.

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

This fixes imported accounts getting deleted when the vault is recreated 

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #1105

### CODE REVIEW

File: app/components/Views/ChoosePassword/index.js [hunk 1]
```diff
@@ -355,6 +355,18 @@ class ChoosePassword extends PureComponent {
 	recreateVault = async password => {
 		const { KeyringController, PreferencesController } = Engine.context;
 		const seedPhrase = await this.getSeedPhrase();
+
+		const importedAccounts = [];
+
+		// Get imported accounts
+		if (KeyringController.state.keyrings.length === 2) {
```
Comment (Gudahtt) [inline app/components/Views/ChoosePassword/index.js:362]: On the extension we have one keyring for each imported account - I am guessing mobile is the same? So this should handle all keyrings from index 1 onwards.

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Few issues

**Issue 1:**   Wallet with 1 imported account produces an error "Error: Invalid password"

**Issue 2:** Wallet with 2 imported accounts does allow you create password, however, the imported accounts are lost

Here goes my set up/steps:

1.  I built v0.2.19 

Screen Shot 2020-08-07 at 6 31 59 PM

2.  I then created 3 wallets (no password set/nor no back up)
wallet 1 = 1 account with 1 imported account
wallet 2 = 1 account with 2 imported accounts 
wallet 3 = 3 accounts with 1 imported account

image

3.   I then switched to this branch and ran yarn clean, yarn watch:clean, and yarn start:android

4.  Upon building this branch, after being prompted to create a password these are the results; seen here = https://recordit.co/189ncJf7qX

wallet 1 = upon attempting to create a password it gave me the "Invalid password" error
wallet 2 = it did allow me to create password, however, my imported accounts are no longer there
wallet 3 = same error as wallet 1

Screen Shot 2020-08-07 at 6 52 19 PM

**Comment by rickycodes** [comment]
> Issue 2: Wallet with 2 imported accounts does allow you create password, however, the imported accounts are lost

I believe this is due to what @Gudahtt mentioned here: https://github.com/MetaMask/metamask-mobile/pull/1756#discussion_r467210676

File: app/components/Views/ChoosePassword/index.js (hunk 1 above)
Comment (andrepimenta) [reply app/components/Views/ChoosePassword/index.js]: You are right! Thanks, fixing this now.

Comment (andrepimenta) [reply app/components/Views/ChoosePassword/index.js]: Should be fixed now 👍

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
**Issue 1** has been fixed 👍  in regards to no longer showing the error on password creation
**Issue 2** still happens to both wallets with 1 imported account and wallets with 2 or more imported accounts

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good, QA Passed 👍

File: app/components/Views/ChoosePassword/index.js [hunk 2]
```diff
@@ -355,6 +355,25 @@ class ChoosePassword extends PureComponent {
...
+			);
+			for (let i = 0; i < simpleKeyrings.length; i++) {
+				const simpleKeyring = simpleKeyrings[i];
+				const simpleKeyringAccounts = await Promise.all(
+					simpleKeyring.accounts.map(account => KeyringController.exportAccount(keychainPassword, account))
+				);
+				importedAccounts = [...importedAccounts, ...simpleKeyringAccounts];
+			}
+		} catch (e) {
+			console.warn(e);
```
Comment (estebanmino) [inline app/components/Views/ChoosePassword/index.js:374]: what if we use the logger here to get the errors in sentry?

**Pull request merged**